contracts/token/ERC721/ERC721Token.sol

```diff
@@ -1,214 +1,99 @@
+  string private _name;
 
-  // Mapping from token ID to approved address
-  mapping (uint256 => address) private tokenApprovals;
+  // Token symbol
+  string private _symbol;
 
-  // Mapping from owner to list of owned token IDs
-  mapping (address => uint256[]) private ownedTokens;
-
-  // Mapping from token ID to index of the owner tokens list
-  mapping(uint256 => uint256) private ownedTokensIndex;
-
-  /**
-  * @dev Guarantees msg.sender is owner of the given token
-  * @param _tokenId uint256 ID of the token to validate its ownership belongs to msg.sender
-  */
-  modifier onlyOwnerOf(uint256 _tokenId) {
-    require(ownerOf(_tokenId) == msg.sender);
-    _;
-  }
+  // Mapping from owner to operator approvals
+  mapping (address => mapping (address => bool)) private operatorApprovals;
 
   /**
-  * @dev Gets the total amount of tokens stored by the contract
-  * @return uint256 representing the total amount of tokens
+  * @dev Constructor function
   */
-  function totalSupply() public view returns (uint256) {
-    return totalTokens;
+  function ERC721Token(string name, string symbol) public {
+    _name = name;
+    _symbol = symbol;
   }
 
   /**
-  * @dev Gets the balance of the specified address
-  * @param _owner address to query the balance of
-  * @return uint256 representing the amount owned by the passed address
+  * @dev Gets the token name
+  * @return string representing the token name
   */
-  function balanceOf(address _owner) public view returns (uint256) {
-    return ownedTokens[_owner].length;
+  function name() public view returns (string) {
+    return _name;
   }
 
   /**
-  * @dev Gets the list of tokens owned by a given address
-  * @param _owner address to query the tokens of
-  * @return uint256[] representing the list of tokens owned by the passed address
+  * @dev Gets the token symbol
+  * @return string representing the token symbol
   */
-  function tokensOf(address _owner) public view returns (uint256[]) {
-    return ownedTokens[_owner];
+  function symbol() public view returns (string) {
+    return _symbol;
   }
 
   /**
-  * @dev Gets the owner of the specified token ID
-  * @param _tokenId uint256 ID of the token to query the owner of
-  * @return owner address currently marked as the owner of the given token ID
+  * @dev Gets the token ID at a given index of the tokens list of the requested owner
+  * @param _owner address owning the tokens list to be accessed
+  * @param _index uint256 representing the index to be accessed of the requested tokens list
+  * @return uint256 token ID at the given index of the tokens list owned by the requested address
   */
-  function ownerOf(uint256 _tokenId) public view returns (address) {
-    address owner = tokenOwner[_tokenId];
-    require(owner != address(0));
-    return owner;
+  function tokenOfOwnerByIndex(address _owner, uint256 _index) public view returns (uint256) {
+    require(_index < balanceOf(_owner));
+    return tokensOf(_owner)[_index];
```

Comment:
This is an order of magnitude slower as it returns the entire array and then indexes the array, and would not work on large arrays. A better way would to make ownedTokens internal and use:

return ownedTokens[_owner][_index];

Reply:
Already on it! https://github.com/spalladino/zeppelin-solidity/commit/b81b4e145e5f65fb3057b035b70edfe9d78d1b6d#diff-bc9f12d1d1e8ef6bdba9bc4d73ae8329R57